File: evidence/generated-systems/django/ecommerce-on-django/ecommerceOnDjango/models/CartItem.py
from django.db import models

#======================================================================
# 
# Encapsulates data for model CartItem
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class CartItem Declaration
#======================================================================
class CartItem (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	quantity = models.IntegerField(null=True)
	unitPrice = Money
	totalPrice = Money
	cart = models.ForeignKey('Cart', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	variant = models.ForeignKey('ProductVariant', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	appliedPromotions = models.ManyToManyField('Promotion',  blank=True, related_name='+')

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.quantity
		str = str + self.unitPrice
		str = str + self.totalPrice
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "CartItem";
    
	def objectType(self):
		return "CartItem";

File: evidence/generated-systems/rubyonrails/ecommerce-on-rails/ecommerceonrails/app/models/cartitem.rb
class CartItem < ApplicationRecord


  composed_of :money,
    class_name: "Money",
    mapping: [
      ${$mapping}, 
      %w[money_currency currency]
    ]

  composed_of :money,
    class_name: "Money",
    mapping: [
      ${$mapping}, 
      %w[money_currency currency]
    ]

  has_many :Cart, class_name: 'Cart'
  has_many :Variant, class_name: 'ProductVariant'
  has_many :AppliedPromotions, class_name: 'Promotion'

end
